app/controllers/katello/api/v2/upstream_subscriptions_controller.rb

```diff
@@ -27,10 +27,31 @@ def index
       respond(collection: collection)
     end
 
+    api :POST, "/organizations/:organization_id/upstream_subscriptions",
+      N_("Add subscriptions consumed by a manifest from Red Hat Subscription Management")
+    param :organization_id, :number, desc: N_("Organization ID"), required: true
+    param :pools, Array, desc: N_("Array of pool IDs to add"), required: true do
+      param :id, String, desc: N_("Pool ID"), required: true
+      param :quantity, :number, desc: N_("Quantity of entitlements to bind")
+    end
+    def create
```

Comment:
I was thinking about failing in the controller action if entitlements were empty for my related PR. Do you think it's better to fail here if we don't have enough data to complete the action, or let the action fail on its own?

Reply:
I'd say that failing in an action gives more data back to the user about what went wrong and also serves as a paper trail.